src/deepsparse/loggers/metric_functions/registry.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Registry that relates the task and the data logging identifier to the
+set of relevant built-in functions
+"""
+from collections import defaultdict
+
+
+__all__ = ["DATA_LOGGING_REGISTRY", "register"]
+
+
+DATA_LOGGING_REGISTRY = defaultdict(lambda: defaultdict(str))
+
+
+def register(task: str, identifier: str):
```

Comment:
might want to rename task to "group" so we can eventually register multiple "groups" for a task